test/support/stripe_case.ex

```diff
@@ -5,19 +5,74 @@ defmodule Stripe.StripeCase do
 
   use ExUnit.CaseTemplate
 
-  def assert_stripe_requested(_method, _url, _extra \\ []) do
-    # TODO: use something akin to WebMock to check the API calls are correct
-    assert true
+  def assert_stripe_requested(expected_method, path, extra \\ []) do
+    expected_url = build_url(path, Keyword.get(extra, :query))
+    expected_body = Keyword.get(extra, :body)
+    expected_headers = Keyword.get(extra, :headers)
+
+    assert_received({method, url, headers, body, _})
```

Comment:
Did you mean for these to be prefixed with `expected_`?

Reply:
No, they will pattern match to the received message. So `url` will contain the url the http client used. Right?

Reply:
oh i figured these had to be variables in this function's scope.  Not seeing 'method' in this scope for example.

Reply:
They could've been pinned like `assert_received({^expected_method, ^expected_url, headers, body, _})` as well.

Reply:
Ah it is a macro that is expecting to pattern match the received message into those variables, correct? Now I understand.